function.annouce.validate.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+function validate_ipv6() {
+	// If we get an IPv6 parameter, use that, else, use client's address
+	// The IPv6 address can either be in the form of:
+	// dead:beef::1234 - i.e., raw address, in which case we use port=
+	// -or-
+	// [dead:beef::1234]:12345 - i.e., enbedded port. Unfortunately
```

Comment:
This case was coded, but didn't work due to a typo+missing return. Comment updated to clarify that this codepath is now properly handled, as well as tested.